dist/tools/kconfiglib/README.md

```diff
@@ -0,0 +1,21 @@
+# kconfiglib
+
+## Examples
+
+The examples folder contains examples of design patterns that can be used along
+with some information of how and when to use them.
+
+## Symbol Conventions
+
+- `ERROR_*` - If this symbol is selected, it will throw an error in the build system to indicate something is configured wrong. Used when the base kconfig tool is not expressive enough.
+- `HAS_*` - Used for modeling the `FEATURES_PROVIDED` functionality that exists in the `make` build system.
+- `HAVE_*` - Used for provided features that do not exist in the `make` build system.
+- `MODULE_*` - if using kconfig dependency resolution, will export to `USEMODULE`
+in make.
+- `USEMODULE_*` - imports the make based dependency resolution to of `USEMODULE` to kconfig and is only used for configuration settings.
+
+## FAQs
+
+### What are choice overrides only allowed in boards and applications?
```

Comment:
Why are we starting this separately from the normal documentation? I'm pretty sure you can fit a FAQs section in the Kconfig doxygen page.